Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs

```diff
@@ -3086,43 +3228,90 @@ specialZoneFlow estimatedQuotes state = do
     _ <- pure $ toast (getString NO_DRIVER_AVAILABLE_AT_THE_MOMENT_PLEASE_TRY_AGAIN)
     continue state { props {currentStage = SearchLocationModel}, data{currentSearchResultType = QUOTES}}
 
-estimatesListFlow :: Array EstimateAPIEntity -> HomeScreenState -> Eval Action ScreenOutput HomeScreenState
-estimatesListFlow estimates state = do
-  let newState = if state.props.isRepeatRide && isJust state.props.repeatRideServiceTierName && not checkRecentRideVariantInEstimates estimates state.props.repeatRideServiceTierName
-                         then state { props {isRepeatRide = false}}
-                         else state
-      estimatesInfo = getEstimatesInfo estimates "" newState
-  if ((not (null estimatesInfo.quoteList)) && (isLocalStageOn FindingEstimate)) then do
-    let _ = unsafePerformEffect $ logEvent state.data.logField "ny_user_quote"
-        nearByDrivers = getNearByDrivers estimates
-        nearByDriversLength = length nearByDrivers
-    _ <- pure $ updateLocalStage SettingPrice
+estimatesListFlow :: Array EstimateAPIEntity -> HomeScreenState -> Int -> Eval Action ScreenOutput HomeScreenState
+estimatesListFlow estimates state count = do
+  let 
+    repeatRideFailCheck = isJust state.props.repeatRideServiceTierName && not checkRecentRideVariantInEstimates estimates state.props.repeatRideServiceTierName -- check if the repeat ride variant is available in the estimates
```

Comment:
Please check these cases And
Remove comments if resolved (Everywhere in this file)

Reply:
added comments for Readability

Reply:
Not talking about this
We so many comments as TODO in the PR 
please check those